src/scene/assetmanager.cpp

```diff
@@ -622,6 +623,27 @@ void AssetManager::setUpAssetLuaTable(Asset* asset) {
     );
     lua_setfield(*_luaState, assetTableIndex, "export");
 
+    // Register registerIdentifierWithMeta function to add meta at runtime
+    // registerIdentifierWithMeta(string identifier)
+    ghoul::lua::push(*_luaState, this, asset);
+    lua_pushcclosure(
+        *_luaState,
+        [](lua_State* L) {
+            ZoneScoped
+
+            Asset* asset = ghoul::lua::userData<Asset>(L, 2);
+            ghoul::lua::checkArgumentsAndThrow(L, 1, "lua::registerIdentifierWithMeta");
+
+            std::string identifier = luaL_checkstring(L, 1);
```

Comment:
->  `std::string identifier = ghoul::lua::value<std::string>(L, 1);`

Then you don't need the `lua_settop(L, 0)` anymore since the `value` function has an argument that determines whether it should be popped or not